framework/JobHandler.py

```diff
@@ -401,20 +418,29 @@ def isFinished(self):
       @ In, None
       @ Out, isFinished, bool, True all the runs in the queue are finished
     """
+    tempList=copy.copy(self.__running+self.__clientRunning)
+    len1 = copy.deepcopy(len(self.__queue))
+    len2 = copy.deepcopy(len(self.__clientQueue))
+
+    import time
+    time.sleep(0.001)
```

Comment:
Add fixme and some comments about these two line. Some suggestion, integrate with RAVEN sleep time in steps.

Reply:
An extensive comment has been added in the text and the issue has been opened (see #1402 )